app/src/ui/local-changes-overwritten/local-changes-overwritten-dialog.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react'
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DefaultDialogFooter,
+} from '../dialog'
+import { OkCancelButtonGroup } from '../dialog/ok-cancel-button-group'
+import { Repository } from '../../models/repository'
+import { RetryAction } from '../../models/retry-actions'
+import { Dispatcher } from '../dispatcher'
+
+interface ILocalChangesOverwrittenDialogProps {
+  readonly repository: Repository
+  readonly dispatcher: Dispatcher
+  /**
+   * Whether there's already a stash entry for the local branch.
+   */
+  readonly hasExistingStash: boolean
+  /**
+   * The action that should get executed if the user selects "Stash and Continue".
+   */
+  readonly retryAction: RetryAction
+  /**
+   * Callback to use when the dialog gets closed.
+   */
+  readonly onDismissed: () => void
+}
+interface ILocalChangesOverwrittenDialogState {
+  readonly loading: boolean
+}
+
+export class LocalChangesOverwrittenDialog extends React.Component<
+  ILocalChangesOverwrittenDialogProps,
+  ILocalChangesOverwrittenDialogState
+> {
+  public constructor(props: ILocalChangesOverwrittenDialogProps) {
+    super(props)
+    this.state = { loading: false }
+  }
+
+  public render() {
+    return (
+      <Dialog
+        title="Error"
+        loading={this.state.loading}
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.onSubmit}
+        type="error"
+      >
+        <DialogContent>
+          <p>
+            Unable to perform this action when changes are present on your
```

Comment:
"this action" is a bit too generic for my taste. Could we attempt to look at the retryAction type and say something like "Unable to merge when changes..." and fall back to "perform this action" only if the retryAction isn't something we're expecting?

Reply:
I like it! Going to do that :)